src/Mod/Start/App/ThumbnailSource.cpp

```diff
@@ -0,0 +1,213 @@
+        const auto f3d = QString::fromUtf8(hGrp->GetASCII("f3d", "f3d").c_str());
+        QStringList args(_f3d.baseArgs);
+        args << QLatin1String("--output=") + _thumbnailPath << _file;
+
+        _process = std::make_unique<QProcess>();
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+        _finishedConnection =
+            QObject::connect(_process.get(),
+                             &QProcess::finished,
+                             /*ThumbnailSource is not derived from QObject, so wrap the signal in a
+                               lambda to propagate it.*/
+                             [this](int exitCode, QProcess::ExitStatus exitStatus) {
+                                 this->f3dRunFinished(exitCode, exitStatus);
+                             });
+#else
+        // In Qt5 we need to resolve the overload of the `finished` signal:
+        _finishedConnection =
+            QObject::connect(_process.get(),
+                             QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),
+                             /*ThumbnailSource is not derived from QObject, so wrap the signal in a
+                               lambda to propagate it.*/
+                             [this](int exitCode, QProcess::ExitStatus exitStatus) {
+                                 this->f3dRunFinished(exitCode, exitStatus);
+                             });
+#endif
+        Base::Console().Log("Creating thumbnail for %s...\n", _file.toStdString());
+        _process->start(f3d, args);
+    }
+}
+
+std::tuple<int, int, int> extractF3DVersion(const QString& stdoutString)
+{
+    int major {0};
+    int minor {0};
+    int patch {0};
+    for (auto lines = stdoutString.split(QLatin1Char('\n')); const auto& line : lines) {
+        if (line.startsWith(QLatin1String("Version: "))) {
+            const auto substring = line.mid(8);
+            if (auto split = substring.split(QLatin1Char('.')); split.size() >= 3) {
+                try {
+                    major = split[0].toInt();
+                    minor = split[1].toInt();
+                    patch = split[2].toInt();
+                }
+                catch (...) {
+                    Base::Console().Log(
+                        "Could not determine F3D version, disabling thumbnail generation\n");
+                }
+            }
+            break;
+        }
+    }
+    return std::make_tuple(major, minor, patch);
+}
+
+QStringList getF3DOptions(const QString& f3d)
+{
+    // F3D is under active development, and the available options change with some regularity.
+    // Rather than hardcode per version, just check the ones we care about.
+    QStringList optionsToTest {
+        QStringLiteral("--load-plugins=occt"),
+        QStringLiteral("--config=thumbnail"),
+        QStringLiteral("--verbose=quiet"),
+        QStringLiteral("--resolution=256,256"),
+        QStringLiteral("--filename=0"),
+        QStringLiteral("--grid=0"),

```

Comment:
```suggestion
        QStringLiteral("--grid=0"),
        QStringLiteral("--axis=0"),
```
was missing from my previous suggestion